cupy/core/_routines_statistics.pyx

```diff
@@ -325,19 +328,30 @@ cpdef ndarray _median(
     else:
         kth = [(sz - 1) // 2]
 
-    if overwrite_input:
+    if a.dtype.char == 'e':
+        # partition doesn't support float16, so temporarily convert to float32
+        part = a.astype(cupy.float32)
```

Comment:
how hard would be to modify partition to support float16?

Reply:
Currently the failure for float16 is in `thrust.sort` which gets called here under certain conditions:
https://github.com/cupy/cupy/blob/49fc35379f17b056a0f311992a85ccb639a5953c/cupy/core/_routines_sorting.pyx#L178-L180

I added a new commit here that moves the float32 conversion into this code branch of `partition` instead of `median`. I didn't try looking at the code in thrust itself.

Reply:
Thanks!!